pattern.c

```diff
@@ -237,6 +238,80 @@ static bool eat_regex(struct Pattern *pat, struct Buffer *s, struct Buffer *err)
   return true;
 }
 
+/**
+ * eat_query - Parse a query for an external search program
+ * @param pat  Pattern to match
+ * @param s   String to parse
+ * @param err Buffer for error messages
+ * @retval true If the pattern was read succesfully
+ */
+static bool eat_query(struct Pattern *pat, struct Buffer *s, struct Buffer *err)
+{
+  struct Buffer cmd_buf;
+  struct Buffer tok_buf;
+  int result;
+  FILE *fp = NULL;
+  pid_t pid;
+  char* oldline = NULL;
+  char* newline;
+  char* dupline;
+  char* nows;
```

Comment:
Initialise these pointers to `NULL`

Reply:
It won't hurt, though it won't help either.  Consider it done.

Reply:
Hi, I pushed an update that addresses the style comments, with the help of the new file iterators.  Please ACK; after that I will go on to the more substantial changes requested by @flatcap .